src/descriptor/bare.rs

```diff
@@ -258,30 +231,38 @@ impl<Pk: MiniscriptKey> Pkh<Pk> {
     }
 }
 
+impl<Pk: MiniscriptKey> Pkh<Pk> {
+    /// Computes an upper bound on the weight of a satisfying witness to the
+    /// transaction.
+    ///
+    /// Assumes all ec-signatures are 73 bytes, including push opcode and
+    /// sighash suffix. Includes the weight of the VarInts encoding the
+    /// scriptSig and witness stack length.
+    pub fn max_satisfaction_weight(&self) -> usize {
+        4 * (1 + 73 + BareCtx::pk_len(&self.pk))
+    }
+}
+
 impl<Pk: MiniscriptKey + ToPublicKey> Pkh<Pk> {
-    /// Obtain the corresponding script pubkey for this descriptor
-    /// Non failing verion of [`DescriptorTrait::script_pubkey`] for this descriptor
-    pub fn spk(&self) -> Script {
-        let addr = bitcoin::Address::p2pkh(&self.pk.to_public_key(), bitcoin::Network::Bitcoin);
+    /// Obtains the corresponding script pubkey for this descriptor.
+    pub fn script_pubkey(&self) -> Script {
+        let addr = Address::p2pkh(&self.pk.to_public_key(), Network::Bitcoin);
         addr.script_pubkey()
```

Comment:
This could delegate to `address` internally.

Reply:
Cool, done. Added the comment about using a hard coded address from the discussion above also.